### PR TITLE
Fix for Dart 2 runtime + 2.0.0-dev.61.0

### PR DESCRIPTION
**Pull request by srawlins**
This is not the last fix for Dart 2 runtime mode, so I don't recommend releasing yet.

### CODE REVIEW

File: packages/file/lib/src/backends/record_replay/recording_proxy_mixin.dart [hunk 1]
```diff
@@ -90,6 +90,10 @@ abstract class RecordingProxyMixin implements ProxyObject, ReplayAware {
   @protected
   Stopwatch get stopwatch;
 
+  // This check is used in noSuchMethod to detect if this code is running in a
+  // Dart 1 runtime, or Dart 2.
+  bool get _runningDart1Runtime => <dynamic>[] is List<String>;
```
Comment (tvolkert) [inline packages/file/lib/src/backends/record_replay/recording_proxy_mixin.dart]: Can you add a TODO referencing dart-lang/sdk#33459 to remove this once the underlying issue is fixed?

File: packages/file/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+#### dev
```
Comment (tvolkert) [inline packages/file/CHANGELOG.md]: Do you not need to publish a new version right away?

File: packages/file/lib/src/backends/record_replay/recording_proxy_mixin.dart (hunk 1 above)
Comment (srawlins) [reply packages/file/lib/src/backends/record_replay/recording_proxy_mixin.dart]: This one actually doesn't have to do with https://github.com/dart-lang/sdk/issues/33459. But I can add a TODO that we can remove once the minimum Dart SDK version is... 2.0 or something.

File: packages/file/CHANGELOG.md (hunk 2 above)
Comment (srawlins) [reply packages/file/CHANGELOG.md]: No. I need workarounds or something for https://github.com/dart-lang/sdk/issues/33459 in addition to this.

Comment (srawlins) [reply packages/file/CHANGELOG.md]: Which I intend to write. Tonight or tomorrow.

File: packages/file/lib/src/backends/record_replay/recording_proxy_mixin.dart (hunk 1 above)
Comment (srawlins) [reply packages/file/lib/src/backends/record_replay/recording_proxy_mixin.dart]: Done.

**Comment by srawlins** [comment]
OK Ready to merge!

**Pull request merged**